lib/users/routes.js

```diff
@@ -71,16 +92,126 @@ TODO: inject content into form
 @function
 @name /users/edit/{{id}}
 */
-router.get('/edit/:id', (req, res) => {
-  // this render should populate the form
-  res.render('users/add', {
-    pageTitle: `${config.users.usersHome.title} :: Edit User`,
+router.get('/edit/:id',
+  loginMiddleware({
+    'role': 'superadmin',
+    'redirectTo': '/login',
+    'setReturnTo': '/users/add',
+  }),
+  (req, res, next) => {
+    req.session.editUserId = req.params.id; // eslint-disable-line no-param-reassign
+
+    database('users').where('id', req.params.id)
+    .then(rows => {
+      if (rows.length < 1) {
+        const err = new Error(`User with id ${req.params.id} does not exist`);
+        err.status = 404;
+        res.render('error', {
+          message: err.message,
+          error: err,
+        });
+      }
+
+      users().then(userModel => {
+        // remove passwoord
+        // TODO: password manipulation
+        userModel[0].attributes.splice(1, 1);
+
```

Comment:
No error checking for multiple rows.

Either check for multiple and throw an error, or save the single item for reuse instead of doing `rows[0]` everywhere